### PR TITLE
Bugfix correct error handling for agent apis

### PR DESCRIPTION
**Pull request by kpango**
### Description:
agent component returns internal server error, but some case it should be return NotFound, AlreadyExists, etc...

I added 4 changes.
- agent vqueue api for correct not found control
- agent readyForUpdate function return value changed from boolean to error type
- add gRPC status convertor to internal/trace package
- add gRPC error status parser

### Environment:

- Go Version: 1.16
- Docker Version: 19.03.8
- Kubernetes Version: 1.18.2
- NGT Version: 1.12.3

### Types of changes:

- [x] Bug fix [type/bug]
- [ ] New feature [type/feature]
- [ ] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [x] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you successfully ran tests with your changes locally?

### Checklist:

- [ ] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [ ] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [ ] I have added tests and benchmarks to cover my changes.
- [ ] I have ensured all new and existing tests passed.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by vankichi — DISMISSED** [review]
LGTM about the point of view of changes test file

File: pkg/agent/core/ngt/handler/grpc/handler.go [hunk 1]
```diff
@@ -443,20 +446,60 @@ func (s *server) Insert(ctx context.Context, req *payload.Insert_Request) (res *
...
+				&errdetails.ResourceInfo{
+					ResourceType: ngtResourceType + "/ngt.Insert",
+					ResourceName: s.ip,
+					Owner:        errdetails.ValdResourceOwner,
+					Description:  err.Error(),
+				})
+			log.Warn(err)
+			code = trace.StatusCodeAlreadyExists(err.Error())
+		} else if errors.Is(err, errors.ErrUUIDNotFound(0)) {
+			err = status.WrapWithInvalidArgument(fmt.Sprintf("Insert API invalid uuid \"%s\" detected", vec.GetId()), err,
```
Comment (rinx) [inline pkg/agent/core/ngt/handler/grpc/handler.go:465]: Maybe better to tell "uuid is empty".

```suggestion
			err = status.WrapWithInvalidArgument(fmt.Sprintf("Insert API empty uuid \"%s\" was given", vec.GetId()), err,
```

Comment (kpango) [reply pkg/agent/core/ngt/handler/grpc/handler.go:465]: Okay

**Review by rinx — APPROVED** [review]

**Comment by kpango** [comment]
I'll merge this PR after CI finished. Thank you for your cooperation

**Pull request merged**